crates/cli-support/src/js/mod.rs

```diff
@@ -1314,6 +1309,7 @@ impl<'a> Context<'a> {
                     const ret = encodeString(arg, view);
                     {debug_end}
                     offset += ret.written;
+                    ptr = realloc(ptr, len, offset, 1) >>> 0;
```

Comment:
Could add a check for whether `len != offset` before reallocing again, to avoid spurious allocs.

Reply:
I considered that, but I didn't think it'd be worth it since in the vast majority of cases the string won't take up the maximum amount of space anyway.

I'd be fine to add it if you want but I don't think the difference would be significant either way.